ARCs/arc-0072.md

```diff
@@ -0,0 +1,341 @@
+status: Draft
+type: Standards Track
+category: Interface
+created: 2023-01-10
+requires: 3, 4, 22, 28, 73
+---
+
+# Algorand Smart Contract NFT Specification
+
+## Abstract
+This specifies an interface for non-fungible tokens (NFTs) to be implemented on Algorand as smart contracts.
+This interface defines a minimal interface for NFTs to be owned and traded, to be augmented by other standard interfaces and custom methods.
+
+
+## Motivation
+Currently most NFTs in the Algorand ecosystem are implemented as ASAs.
+However, to provide rich extra functionality, it can be desireable to implement NFTs as a smart contract instead.
+To foster an interoperable NFT ecosystem, it is necessary that the core interfaces for NFTs be standardized.
+
+
+## Specification
+The key words "**MUST**", "**MUST NOT**", "**REQUIRED**", "**SHALL**", "**SHALL NOT**", "**SHOULD**", "**SHOULD NOT**", "**RECOMMENDED**", "**MAY**", and "**OPTIONAL**" in this document are to be interpreted as described in <a href="https://www.ietf.org/rfc/rfc2119.txt">RFC-2119</a>.
+
+
+### Core NFT specification
+
+A smart contract NFT that is compliant with this standard must implement the interface detection standard defined in [ARC-73](./arc-0073.md).
+
+Additionally, the smart contract MUST implement the following interface:
+
+```json
+{
+  "name": "ARC-72",
+  "desc": "Smart Contract NFT Base Interface",
+  "methods": [
+    {
+      "name": "ownerOf",
+      "desc": "Returns the address of the current owner of the NFT with the given tokenId",
+      "readonly": true,
+      "args": [
+        { "type": "uint256", "name": "tokenId", "desc": "The ID of the NFT" },
+      ],
+      "returns": { "type": "address", "desc": "The current owner of the NFT." }
+    },
+    {
+      "name": "transferFrom",
+      "desc": "Transfers ownership of an NFT",
+      "readonly": false,
+      "args": [
+        { "type": "address", "name": "from" },
+        { "type": "address", "name": "to" },
+        { "type": "uint256", "name": "tokenId" }
+      ],
+      "returns": { "type": "void" }
+    },
+  ],
+  "events": [
+    {
+      "name": "Transfer",
+      "desc": "Transfer ownership of an NFT",
+      "args": [
+        {
+          "type": "address",
+          "name": "from",
+          "desc": "The current owner of the NFT"
+        },
+        {
+          "type": "address",
+          "name": "to",
+          "desc": "The new owner of the NFT"
+        },
+        {
+          "type": "uint256",
+          "name": "tokenId",
+          "desc": "The ID of the transferred NFT"
+        }
+      ]
+    }
+  ]
+}
+```
+
+Ownership of a token ID by the zero address indicates that ID is invalid.
+The `ownerOf` method MUST return the zero address for invalid token IDs.
+The `transferFrom` method MUST error when `from` is not the owner of `tokenId`, unless called by an approved operator as defined by an extension such as the transfer management extension defined in this ARC.
```

Comment:
This doesn't prevent others from transferring my NFTs. As long as they call the method and specify my address as the first argument, then they could transfer it. What we care about is that the entity *calling* the `transferFrom` method is either the owner or an approved operator.